src/index.d.ts

```diff
@@ -309,7 +309,7 @@ interface Branch {
   getLatestReferringParams: (synchronous?: boolean) => Promise<BranchParams>;
   getFirstReferringParams: () => Promise<BranchParams>;
   lastAttributedTouchData: (attributionWindow?: number) => Promise<BranchParams>;
-  setIdentity: (identity: string) => void;
+  setIdentity: (identity: string) => Promise<BranchParams>;
```

Comment:
Since this changes the type, code that exists today may get a warning since it is no longer void and the return type isn't being used. Can you confirm this doesn't break anything?

Reply:
Would it make sense to keep the void function and add a new function `setIdentityWithPromise` that returns the promise/callback?